sql/core/src/main/scala/org/apache/spark/sql/execution/command/AnalyzeColumnCommand.scala

```diff
@@ -0,0 +1,209 @@
+            colStats += attr.name -> BasicColStats(
+              dataType = attr.dataType,
+              numNulls = rowCount - statsRow.getLong(pos + NumericStatsAgg.numNotNullsIndex),
+              max = Option(statsRow.get(pos + NumericStatsAgg.maxIndex, attr.dataType)),
+              min = Option(statsRow.get(pos + NumericStatsAgg.minIndex, attr.dataType)),
+              ndv = Some(statsRow.getLong(pos + NumericStatsAgg.ndvIndex)))
+            pos += NumericStatsAgg.statsSeq.length
+          case StringType =>
+            colStats += attr.name -> BasicColStats(
+              dataType = attr.dataType,
+              numNulls = rowCount - statsRow.getLong(pos + StringStatsAgg.numNotNullsIndex),
+              maxColLen = Some(statsRow.getLong(pos + StringStatsAgg.maxLenIndex)),
+              avgColLen =
+                Some(statsRow.getLong(pos + StringStatsAgg.sumLenIndex) / (1.0 * rowCount)),
+              ndv = Some(statsRow.getLong(pos + StringStatsAgg.ndvIndex)))
+            pos += StringStatsAgg.statsSeq.length
+          case BinaryType =>
+            colStats += attr.name -> BasicColStats(
+              dataType = attr.dataType,
+              numNulls = rowCount - statsRow.getLong(pos + BinaryStatsAgg.numNotNullsIndex),
+              maxColLen = Some(statsRow.getLong(pos + BinaryStatsAgg.maxLenIndex)),
+              avgColLen =
+                Some(statsRow.getLong(pos + BinaryStatsAgg.sumLenIndex) / (1.0 * rowCount)))
+            pos += BinaryStatsAgg.statsSeq.length
+          case BooleanType =>
+            val numOfNotNulls = statsRow.getLong(pos + BooleanStatsAgg.numNotNullsIndex)
+            val numOfTrues = Some(statsRow.getLong(pos + BooleanStatsAgg.numTruesIndex))
+            colStats += attr.name -> BasicColStats(
+              dataType = attr.dataType,
+              numNulls = rowCount - numOfNotNulls,
+              numTrues = numOfTrues,
+              numFalses = numOfTrues.map(i => numOfNotNulls - i),
+              ndv = Some(2))
+            pos += BooleanStatsAgg.statsSeq.length
+        }
+      }
+      newStats = newStats.copy(basicColStats = colStats.toMap)
+      sessionState.catalog.alterTable(catalogTable.copy(stats = Some(newStats)))
+      // Refresh the cached data source table in the catalog.
+      sessionState.catalog.refreshTable(tableIdent)
+    }
+
+    Seq.empty[Row]
+  }
+
+  private def statsAgg(name: String, dataType: DataType): Seq[Column] = dataType match {
+    // Currently we only support stats generation for atomic types
+    case n: NumericType => NumericStatsAgg(name)
+    case TimestampType | DateType => NumericStatsAgg(name)
+    case StringType => StringStatsAgg(name)
+    case BinaryType => BinaryStatsAgg(name)
+    case BooleanType => BooleanStatsAgg(name)
+    case otherType =>
+      throw new AnalysisException(s"Analyzing column $name of $otherType is not supported.")
+  }
+}
+
+object ColumnStats extends Enumeration {
```

Comment:
I guess it is not encouraged to add `private[sql]` in this case. Please see https://github.com/apache/spark/commit/511f52f8423e151b0d0133baf040d34a0af3d422#diff-04b333ef6416de455e072b2bbbeff9efL37